dotCMS/src/integration-test/java/com/dotcms/experiments/business/IndexRegexUrlPatterStrategyIntegrationTest.java

```diff
@@ -0,0 +1,427 @@
+
+        assertFalse(("https://localhost" + folder.getPath() + "xedni").matches(regexPattern));
+
+        assertFalse(("https://localhost:8080" + folder.getPath() + "indexindex").matches(regexPattern));
+        assertFalse(("https://localhost" + folder.getPath() + "indexindex").matches(regexPattern));
+
+        final String folderPathWithoutSlash = folder.getPath().substring(0, folder.getPath().length() - 1);
+        assertTrue(("https://localhost:8080" + folderPathWithoutSlash).matches(regexPattern));
+        assertTrue(("https://localhost" + folderPathWithoutSlash).matches(regexPattern));
+
+        assertFalse(("https://localhost:8080/any_folder/index").matches(regexPattern));
+        assertFalse(("https://localhost/any_folder/index").matches(regexPattern));
+
+        assertFalse(("https://localhost:8080/index").matches(regexPattern));
+        assertFalse(("https://localhost/index").matches(regexPattern));
+
+        assertFalse(("https://localhost:8080/INDEX").matches(regexPattern));
+        assertFalse(("https://localhost/INDEX").matches(regexPattern));
+
+        assertFalse(("https://localhost:8080/IndEX").matches(regexPattern));
+        assertFalse(("https://localhost/IndEX").matches(regexPattern));
+
+        assertFalse(("https://localhost:8080/").matches(regexPattern));
+        assertFalse(("https://localhost/").matches(regexPattern));
+
+        assertFalse(("https://localhost:8080/aaaa/bbb").matches(regexPattern));
+        assertFalse(("https://localhost:8080/aaaa").matches(regexPattern));
+
+        assertFalse(("https://localhost/aaaa/bbb").matches(regexPattern));
+        assertFalse(("https://localhost/aaaa").matches(regexPattern));
+
+        assertFalse(("https://localhost/xedni").matches(regexPattern));
+
+        assertFalse(("https://localhost:8080/indexindex").matches(regexPattern));
+        assertFalse(("https://localhost/indexindex").matches(regexPattern));
+
+        assertFalse(("https://localhost:8080").matches(regexPattern));
+        assertFalse(("https://localhost").matches(regexPattern));
+        assertFalse(("https://localhost").matches(regexPattern));
+
+        assertTrue(("https://localhost:8080" + folder.getPath() + "/?variantName=dotexperiment-fcaef4575a-variant-1&redirect=true").matches(regexPattern));
+        assertTrue(("https://localhost" + folder.getPath() + "/?variantName=dotexperiment-fcaef4575a-variant-1&redirect=true").matches(regexPattern));
+
+        assertFalse(("https://localhost:8080/?variantName=dotexperiment-fcaef4575a-variant-1&redirect=true").matches(regexPattern));
+        assertFalse(("https://localhost/?variantName=dotexperiment-fcaef4575a-variant-1&redirect=true").matches(regexPattern));
+    }
+
+
+
+    /**
+     * Method to test: {@link IndexRegexUrlPatterStrategy#isMatch(HTMLPageAsset)} and
+     * {@link IndexRegexUrlPatterStrategy#getRegexPattern(HTMLPageAsset)}
+     * When: The page an index page but not the root Index Page and the Domain is not localhost
```

Comment:
The page is an index page*

Reply:
done https://github.com/dotCMS/core/pull/25254/commits/3da387f10dc7fec73ab865b18be94e0739670abb